integration/spark/src/main/java/openlineage/spark/agent/lifecycle/plan/wrapper/InputDatasetVisitor.java

```diff
@@ -0,0 +1,36 @@
+package openlineage.spark.agent.lifecycle.plan.wrapper;
+
+import io.openlineage.client.OpenLineage;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import scala.PartialFunction;
+import scala.runtime.AbstractPartialFunction;
+
+public class InputDatasetVisitor
```

Comment:
I found it clearer when this was logic that was applied to the `Dataset`s returned by the visitor. It's not really a plan visitor- it's a function we apply to the visitors' results. I think we should keep the logic as it was before.

Reply:
not only, you can check OutputDatasetWithMetadataVisitor there is usage of LogicalPlan, not only wrapping a result